woocommerce/woocommerce-accommodation-bookings
# Expose the `accommodation-booking` product type to the REST API

### All Submissions:



---

### Changes proposed in this Pull Request:

The custom product type `accommodation booking` was added to the plugin at:

https://github.com/woocommerce/woocommerce-accommodation-bookings/blob/ad98ad681a8197d202327af7a1d6080952cd1b15/includes/admin/class-wc-accommodation-booking-admin-panels.php#L49

However, when a product is created or updated via the REST API, it throws this error as noted in https://github.com/woocommerce/woocommerce-accommodation-bookings/issues/435#issue-2313206689:

```
"type": "type is not one of simple, grouped, external, variable, and booking."
```

This error indicates that the custom product type `accommodation-booking` is not accessible through the REST API. The problem originates from the following code, which restricts its accessibility:

https://github.com/woocommerce/woocommerce-accommodation-bookings/blob/ad98ad681a8197d202327af7a1d6080952cd1b15/includes/class-wc-accommodation-bookings-plugin.php#L124-L127

This PR addresses the issue by exposing the custom product type `accommodation-booking` to the REST API.

Closes #401.

### Steps to test the changes in this Pull Request:



See the details at https://github.com/woocommerce/woocommerce-accommodation-bookings/issues/435#issue-2313206689 and confirm the issue is resolved.

Note: This PR requires another fix which is made at https://github.com/woocommerce/woocommerce-bookings/pull/3809. Please use them together to test the fix.

### Changelog entry

> Fix - Error in Updating Accommodation Product Data via REST API: Unrecognized Product Type Response
